controllers/imageclusterinstall_controller.go

```diff
@@ -436,24 +430,42 @@ func (r *ImageClusterInstallReconciler) checkClusterTimeout(ctx context.Context,
 	return false, nil
 }
 
-func (r *ImageClusterInstallReconciler) checkClusterStatus(ctx context.Context, log logrus.FieldLogger, ici *v1alpha1.ImageClusterInstall, clusterDeployment *hivev1.ClusterDeployment) (ctrl.Result, error) {
+func (r *ImageClusterInstallReconciler) checkClusterStatus(ctx context.Context,
+	log logrus.FieldLogger,
+	ici *v1alpha1.ImageClusterInstall,
+	bmh *bmh_v1alpha1.BareMetalHost,
+	timedout bool) (ctrl.Result, error) {
 	spokeClient, err := r.spokeClient(ctx, ici)
 	if err != nil {
 		log.WithError(err).Error("failed to create spoke client")
 		return ctrl.Result{}, err
 	}
 
 	if status := r.GetSpokeClusterInstallStatus(ctx, log, spokeClient); !status.Installed {
-		log.Infof("cluster install in progress: %s", status.String())
+		if timedout {
+			r.Log.Infof("cluster install timed out and status is not installed: %s", status.String())
+			return ctrl.Result{}, nil
+		}
+
+		r.Log.Infof("cluster install in progress: %s", status.String())
 		if err := r.setClusterInstallingConditions(ctx, ici, status.String()); err != nil {
-			log.WithError(err).Error("failed to set installing conditions")
+			r.Log.WithError(err).Error("failed to set installing conditions")
 		}
 		return ctrl.Result{RequeueAfter: time.Minute}, nil
 	}
-	log.Info("cluster is installed")
+	r.Log.Info("cluster is installed")
+
+	// After installation ended we don't want that ironic will do any changes in the node
+	patch := client.MergeFrom(bmh.DeepCopy())
```

Comment:
was move here in order to set installed only after setting detached, in case setting detached will fail we will reconcile one more time